src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -458,12 +458,19 @@ private static ImmutableArray<ISymbol> GetSymbolsForNamespaceDeclarationNameCont
                 }
 
                 // If the thing on the left is a lambda expression, we shouldn't show anything.
+                var originalExpressionKind = originalExpression.Kind();
                 if (symbol.Kind == SymbolKind.Method &&
                     ((IMethodSymbol)symbol).MethodKind == MethodKind.AnonymousFunction)
                 {
                     return ImmutableArray<ISymbol>.Empty;
                 }
 
+                if (symbol.Kind == SymbolKind.Method &&
+                    (originalExpressionKind == SyntaxKind.IdentifierName || originalExpressionKind == SyntaxKind.GenericName))
```

Comment:
@sharwell, it seems that I forgot to build project before running tests when I tried to simplify the fix :)

example of why the previous code wouldn't work:
```csharp
class C
{
    string M()
    {
        M().$$
        return string.Empty;
    }
}
```
no suggestion with the previous fix here.